src/AddTodoForm.js

```diff
@@ -1,7 +1,12 @@
-import React, { useState } from "react";
+import React, { useState, useEffect, useRef } from "react";
 
 const AddTodoForm = ({ onAddTodo }) => { // accept onAddTodo as a prop
   const [todoTitle, setTodoTitle] = useState("");
+  const inputRef = useRef(null);
+
+  useEffect (() => {
+    inputRef.current.focus();
+  }, []);
```

Comment:
It looks like you kinda doubled up on the `useRef` logic, and while I see you did create the reusable InputWithLabel component, you haven't imported it here. I think some previous deviations from the curriculum are working against you now, even though they worked fine when you implemented them. As sort of a reset I'm just going to show you an example of how some of these files should look to get you back on track.
```
import React, { useState } from 'react';
import InputWithLabel from './InputWithLabel';

export default function AddTodoForm({ onAddTodo }) {
  const [todoTitle, setTodoTitle] = useState('');

//originally you doing this directly in the input like this: onChange((e)=>setTodoTitle(e.target.value)) 
//which is totally valid, but now that things are moving around it probably makes more sense to
//define and pass this handler into InputWithLabel instead of the state setter.
  function handleTitleChange(e) {
    setTodoTitle(e.target.value);
  }

  function handleAddTodo(e) {
    e.preventDefault();
    onAddTodo({ title: todoTitle, id: Date.now() });
    setTodoTitle('');
  }

  return (
    <form onSubmit={handleAddTodo}>
      <InputWithLabel
        todoTitle={todoTitle}
        handleTitleChange={handleTitleChange}
      >
       // wrapping the component around the children prop is what allows us to 
       // put whatever we want in in, so you don't need to add title to the component itself
        Title
      </InputWithLabel>
      <button type="submit">Add</button>
    </form>
  );
}
```